easybuild/easyblocks/s/star_ccm.py

```diff
@@ -0,0 +1,114 @@
+##
+# Copyright 2018-2019 Ghent University
+#
+# This file is part of EasyBuild,
+# originally created by the HPC team of Ghent University (http://ugent.be/hpc/en),
+# with support of Ghent University (http://ugent.be/hpc),
+# the Flemish Supercomputer Centre (VSC) (https://www.vscentrum.be),
+# Flemish Research Foundation (FWO) (http://www.fwo.be/en)
+# and the Department of Economy, Science and Innovation (EWI) (http://www.ewi-vlaanderen.be/en).
+#
+# https://github.com/easybuilders/easybuild
+#
+# EasyBuild is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation v2.
+#
+# EasyBuild is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with EasyBuild.  If not, see <http://www.gnu.org/licenses/>.
+##
+"""
+EasyBuild support for building and installing STAR-CCM+, implemented as an easyblock
+"""
+import glob
+import os
+import tempfile
+
+import easybuild.tools.environment as env
+from easybuild.framework.easyblock import EasyBlock
+from easybuild.tools.build_log import EasyBuildError
+from easybuild.tools.filetools import change_dir
+from easybuild.tools.run import run_cmd
+
+
+class EB_STAR_minus_CCM_plus_(EasyBlock):
+    """Support for building/installing STAR-CCM+."""
+
+    def __init__(self, *args, **kwargs):
+        """Initialise STAR-CCM+ easyblock."""
+        super(EB_STAR_minus_CCM_plus_, self).__init__(*args, **kwargs)
+        self.starccm_subdir = None
+        self.starview_subdir = None
+
+    def configure_step(self):
+        """No configuration procedure for STAR-CCM+."""
+        pass
+
+    def build_step(self):
+        """No build procedure for STAR-CCM+."""
+        pass
+
+    def install_step(self):
+        """Custom install procedure for STAR-CCM+."""
+        cands = glob.glob("./STAR-CCM+%s_*.sh" % self.version)
+        if len(cands) == 1:
+            install_script = cands[0]
+
+            env.setvar('CHECK_DISK_SPACE', 'OFF')
```

Comment:
that's neat

Reply:
Yeah, and it gets better...

I just spent a couple of hours figuring out why the install script for STAR-CCM+ 16.06.008 was exiting with a non-zero exit code, with zero output.

After finding the log file (buried in `$EBROOTSTARMINCCMPLUS/16.06.008/STAR-CCM+16.06.008/STAR-CCM+16.06.008_InstallLog.log` (yes, the version subdir is there twice...), and mostly seeing `Status: SUCCESSFUL`, I was even more confused.

There was one `Status: ERROR` though, in the disk space check step....

So it seems that setting `$CHECK_DISK_SPACE` to `OFF` skips the disk space check, but also marks that step as failed, and hence the install script exists with a non-zero exit code. 🤦 

So to fix: either don't set `$CHECK_DISK_SPACE` (if that works, I think it's only required when installing to an NFS mounted filesystem), or totally ignore the exit code of the install script and fully rely on the sanity check done by EasyBuild...

Reply:
I disabled this again, since it was causing trouble.

People can still resort to `CHECK_DISK_SPACE=OFF eb ...` when installing STAR-CCM+, if the disk space check is causing trouble.